### PR TITLE
Do not sneak in another dash/plus in format_diff_header

### PR DESCRIPTION
**Pull request by blueyed**
And especially do not match with `*` for this, which might trigger false
positives.

This moves the removal of leading +/- down to where those lines are
printed, instead of applying it globally to the input.

Ref: https://github.com/so-fancy/diff-so-fancy/commit/2f743f07941e744b2f24218dc422f26512823549#commitcomment-16952151

No new tests, but maybe I can get one for 10 innernet points from @scottchiefbaker? :) (https://github.com/so-fancy/diff-so-fancy/commit/2f743f07941e744b2f24218dc422f26512823549#commitcomment-16953657)

### CODE REVIEW

**Comment by scottchiefbaker** [comment]
Before we land this let's wait and see what @pauley-unsaturated says. This would revert part of his code, and un-fix whatever he was fixing. Also nothing is "broken" we just work around it in a different way so it's not really a big deal either way.

File: lib/diff-so-fancy.pl [hunk 1]
```diff
@@ -135,6 +130,10 @@
 	# Just a regular line, print it out #
 	#####################################
 	} else {
+		# Remove the correct number of leading " " or "+" or "-"
+		if ($strip_leading_indicators) {
+			$line = strip_leading_indicators($line,$columns_to_remove);
+		}
```
Comment (scottchiefbaker) [inline lib/diff-so-fancy.pl]: If we're going to move the pre-processing logic for `strip_leading_indicators()` down here we should move `mark_empty_lines()` as well.

**Comment by pauley-unsaturated** [comment]
I'm fine with this.  My original change was to address the fact that we were removing one of the three '+' signs that git diff inserted.  The original matching expression didn't appear to be removing one of them on purpose.  Two '+' or '-' is just fine with me.

File: diff-so-fancy [hunk 2]
```diff
@@ -35,8 +35,8 @@ git_index_line_pattern="^($color_code_regex)index .*"
 format_diff_header () {
   # simplify the unified patch diff header
     $SED -E "/$git_index_line_pattern/{N;s/$git_index_line_pattern\n//;}" \
-    | $SED -E "s/^($color_code_regex)\-\-\-(.*)$/\1$(print_horizontal_rule)\\${NL}\1\-\-\-\-\5/g" \
-    | $SED -E "s/^($color_code_regex)\+\+\+(.*)$/\1\+\+\+\+\5\\${NL}\1$(print_horizontal_rule)/g"
+    | $SED -E "s/^($color_code_regex)\-\-\-(.*)$/\1$(print_horizontal_rule)\\${NL}\1---\5/g" \
+    | $SED -E "s/^($color_code_regex)\+\+\+(.*)$/\1+++\5\\${NL}\1$(print_horizontal_rule)/g"
```
Comment (pauley-unsaturated) [inline diff-so-fancy]: However: does this work? + is a special character in SED.  I guess since this is in the second pattern, kleene plus and kleene star don't apply so they're treated as normal characters?

Comment (scottchiefbaker) [inline diff-so-fancy]: I think as the second parameter there fine because you're not talking about patterns.

Comment (blueyed) [inline diff-so-fancy]: Yes.

File: lib/diff-so-fancy.pl (hunk 1 above)
Comment (blueyed) [inline lib/diff-so-fancy.pl]: Not so easy.. because it also strips escape sequences, and that's expected above already.
I'd say to keep it there for now.

Comment (scottchiefbaker) [inline lib/diff-so-fancy.pl]: Then my vote would be to keep them together for readability. Ultimately `strip_leading_indicators()` is getting called on EVERY output line anyway. Moving it isn't really saving us much in terms of overhead.

Comment (blueyed) [inline lib/diff-so-fancy.pl]: The problem with `strip_leading_indicators` is that it will also strip one level from the `---/+++` markers.  That was the main reason for moving it.

Comment (blueyed) [inline lib/diff-so-fancy.pl]: Okay, I was wrong: it can be moved down.. done in a separate commit.

**Comment by stevemao** [comment]
A test is necessary. Just add what was broken before this change. @blueyed

**Comment by blueyed** [comment]
@stevemao 
I think it's covered with the test in https://github.com/so-fancy/diff-so-fancy/commit/c3a141b7e60c597a38426fa709b6a0e3bd7fe766.
